mlflow/store/artifact/optimized_s3_artifact_repo.py

```diff
@@ -52,7 +53,10 @@ def __init__(
         self._addressing_style = addressing_style
         self._s3_endpoint_url = s3_endpoint_url
         self.bucket, self.bucket_path = self.parse_s3_compliant_uri(self.artifact_uri)
-        self._region_name = self._get_region_name()
+        if region_name is None:
```

Comment:
Dumb question, is passing `region_name` explicitly necessary or if we override `self.get_region_name` in the R2 repo will the correct implementation just get called here? Fine with passing explicitly if needed

Reply:
It's currently a private method in OptimizedS3ArtifactRepository, so there's nothing to override.  Also, this call is happening in the super class's constructor, not w/in the R2ArtifactRepository, so this felt like the fastest route to solving it.